pkg/backend/backend.go

```diff
@@ -245,6 +245,9 @@ type UpdateOptions struct {
 	AutoApprove bool
 	// SkipPreview, when true, causes the preview step to be skipped.
 	SkipPreview bool
+
+	// WatchPaths, allows specifying paths that need to be watched (instead of the whole project).
+	WatchPaths []string
```

Comment:
Instead of putting this on `UpdateOption`, let's put this as a separate argument to `Watch` on the backend interface.  This option isn't relevant to most updates, and is really not about the update part of the watch operation at all - so will be easier to keep it watch-specific.

Reply:
Yeah, I doubted this was the right place to put this in the first place, but said whatever, we can discuss this along the way.

All in all sounds good, will take a look!

Reply:
I added `paths []string` as the last parameter on the `Watch` function.

Had to add it in multiple places. Not sure if I got them all right, but it seems it's good. 

Can you please double-check?

Thank you!